### PR TITLE
Wechaty Chatbot testing subsection of Explanation section

### PR DESCRIPTION
**Pull request by nibble0101**
# Improve the Introduction and Explanation sections project

| Section | Sub-section | Technical Writer |
| --- | --- | --- |
|Explanations | Wechaty chatbot testing |  Joseph Mawa |

This  is **Wechaty chatbot testing** sub-section of the **Explanations** section

image
image
image
image
image
image

### CODE REVIEW

**Review by proudofsimin — APPROVED** [review]
Similar feedback again. Please introduce a list with a complete sentence, instead of a partial one that's completed by the list items. See [Introductory sentences for lists](https://developers.google.com/style/lists#introductory-sentences-for-lists).
(But since this is a minor problem, I'll still approve it.)

**Review by Rohitesh-Kumar-Jain — APPROVED** [review]
Perfect PR :  clean commit history, properly based, relevant screenshots, PR good in scope.

**Comment by Rohitesh-Kumar-Jain** [comment]
@chris-4444 I would like you to once go through the PR, and approve when you think it's ready to get merged.

**Review by chris-4444 — APPROVED** [review]
LGTM.

File: docusaurus/docs/explainations/chatbot-testing.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+---
+title: Wechaty Chatbot testing
```
Comment (lijiarui) [inline docusaurus/docs/explainations/chatbot-testing.md]: We should change to `Testing` here.

**Review by lijiarui — CHANGES_REQUESTED** [review]
I feel a lit bit confuse about this part.

At first, I thought it maybe wechaty testing, but when I dig into it, it seems trying to say chatbot testing, which is unrelated with wechaty. 

@nibble0101 Could you tell me which part reference chatbot testing concept?

Thanks!

File: docusaurus/docs/explainations/chatbot-testing.md (hunk 1 above)
Comment (lijiarui) [reply docusaurus/docs/explainations/chatbot-testing.md]: @chris-4444 @nibble0101 Please change as I request.

Also, I think the file name should change to `testing.md` too.

**Comment by lijiarui** [comment]
I'm not quite sure whether this is really related to wechaty. How do you think of this? @huan

**Comment by huan** [comment]
Chatbot testing is very important for a production environment and enterprise usage.

Wechaty has a testing framework named [wechaty-puppet-mock](https://github.com/wechaty/wechaty-puppet-mock) and a more specific module [wechaty-mocker](https://github.com/wechaty/wechaty-mocker) to help Wechaty developers to do better unit testing for their chatbots.

A conversational testing example can be found at [math_master unit tests](https://github.com/wechaty/wechaty-vorpal-contrib/blob/master/src/contrib/math_master/math_master.spec.ts), I believe we should add that information to this page.

For the other part of this docs page, like the chatbottesting.com etc., I feel they are OK to be there, but should be minor information.

The major information in this page should be how to use **wechaty-puppet-mock** and the **wechaty-mocker**

**Comment by lijiarui** [comment]
Yeah, I just remember we have this:
- https://github.com/wechaty/puppet-services/wiki
- https://github.com/wechaty/PMC/issues/31

File: docusaurus/docs/explainations/chatbot-testing.md (hunk 1 above)
Comment (nibble0101) [reply docusaurus/docs/explainations/chatbot-testing.md]: Changed as requested

File: docusaurus/docs/explainations/testing.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+  /**
+   * Message Processing
+   */
+  user.say("Hello").to(contact);
+  contact.say("World").to(user);
+});
+```
+
+You can read the [unit tests](https://github.com/wechaty/wechaty-vorpal-contrib/blob/master/src/contrib/math_master/math_master.spec.ts) for math_master bot for a full example illustrating the use of `wechaty-puppet-mock` package for conversational testing of your chatbot.
```
Comment (lijiarui) [inline docusaurus/docs/explainations/testing.md:51]: We can add link https://github.com/wechaty/wechaty-vorpal-contrib/tree/master/src/contrib/math_master for math_master bot here.

**Review by lijiarui — CHANGES_REQUESTED** [review]
> For the other part of this docs page, like the chatbottesting.com, etc., I feel they are OK to be there but should be minor information.

I feel that there is too much talking about other testing tools, which is actually unrelated to wechaty testing.

**Review by lijiarui — APPROVED** [review]
LGTM

**Pull request merged**